packages/pipeline-services/src/migration/index.ts

```diff
@@ -19,18 +19,29 @@ import produce from "immer";
 import migrateV1 from "./migrateV1";
 import migrateV2 from "./migrateV2";
 import migrateV3 from "./migrateV3";
+import migrateV4 from "./migrateV4";
 
-export function migrate(pipelineJSON: any) {
+export function migrate(
+  pipelineJSON: any,
+  setNodePathsRelativeToPipeline?: (pipeline: any) => any
+) {
   return produce(pipelineJSON, (draft: any) => {
     const version = draft.pipelines[0].app_data?.version ?? 0;
     if (version < 1) {
+      console.debug("migrating pipeline from v0 to v1");
       migrateV1(draft);
     }
     if (version < 2) {
-      migrateV2(draft);
+      console.debug("migrating pipeline from v1 to v2");
+      migrateV2(draft, setNodePathsRelativeToPipeline);
```

Comment:
I think we already had a conversation about leaving the v2 migration as is. If we want escape hatches for migration we should come up with a more robust API

Reply:
@lresende said the opposite, he wants to make sure we keep all previous functionality for migration, if we dont change this that's a reversion to our current functionality

Reply:
Was this a new discussion? Because we all agreed on this when creating this current migration package